### PR TITLE
feat: monitor the mint balance not greater than the lock balance

### PR DESCRIPTION
**Pull request by pygman**

### CODE REVIEW

File: offchain-modules/packages/app-monitor/src/balanceProvider.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    const searchKey: SearchKey = {
+      script: {
+        code_hash: typescript.codeHash,
+        hash_type: typescript.hashType,
+        args: typescript.args,
+      },
+      script_type: ScriptType.type,
+    };
+    const cells = await this.ckbIndexer.getCells(searchKey, (_index, _cell) => ({ stop: false, push: true }), {
+      sizeLimit: 0x100000,
```
Comment (mkxbl) [inline offchain-modules/packages/app-monitor/src/balanceProvider.ts:66]:
```suggestion
      sizeLimit: 0x1000,
```

File: offchain-modules/packages/app-monitor/src/balanceProvider.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      order: Order.asc,
+    });
+    const balance = cells.map((cell) => BigInt(cell.cell_output.capacity)).reduce((c, c0) => c + c0, BigInt(0));
```
Comment (mkxbl) [inline offchain-modules/packages/app-monitor/src/balanceProvider.ts:69]: sudt balance === capacity ???

File: offchain-modules/packages/app-monitor/src/balanceProvider.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    logger.info(`ckb get_cells token: ${tokenAddress}, cells.length: ${cells.length}, balance: ${balance}`);
```
Comment (mkxbl) [inline offchain-modules/packages/app-monitor/src/balanceProvider.ts:70]: add sudtTypescript to log

File: offchain-modules/packages/app-monitor/src/monitor.ts [hunk 4]
```diff
@@ -328,6 +339,54 @@ export class Monitor {
...
+            .error()
+            .send();
+          logger.error(errorMsg);
+        }
+      },
+      {
+        onRejectedInterval: ForceBridgeCore.config.monitor!.overmintCheckInterval,
+        onResolvedInterval: ForceBridgeCore.config.monitor!.overmintCheckInterval,
+        onRejected: (e: Error) => {
+          logger.error(`Monitor observeEthLock error:${e.stack}`);
```
Comment (mkxbl) [inline offchain-modules/packages/app-monitor/src/monitor.ts:384]:
```suggestion
          logger.error(`Monitor checkOvermintEvent error:${e.stack}`);
```

Comment (pygman) [reply offchain-modules/packages/app-monitor/src/monitor.ts]: fixed

File: offchain-modules/packages/app-monitor/src/balanceProvider.ts (hunk 3 above)
Comment (pygman) [reply offchain-modules/packages/app-monitor/src/balanceProvider.ts]: add sudtTypescript to log

File: offchain-modules/packages/app-monitor/src/balanceProvider.ts (hunk 2 above)
Comment (pygman) [reply offchain-modules/packages/app-monitor/src/balanceProvider.ts]: fixed

File: offchain-modules/packages/app-monitor/src/balanceProvider.ts (hunk 1 above)
Comment (pygman) [reply offchain-modules/packages/app-monitor/src/balanceProvider.ts]: fixed

**Pull request merged**